Award details

peter-budo opened the pull request:
@karllindmark 

Give it a go and let me know if there is something that needs to be changed

peter-budo committed:
[WIP] Award details:
 • Removed dummy strings file

peter-budo committed:
[WIP] Award details:
 • Extracted colour out of styles

peter-budo committed:
[WIP] Award details:
 • Clean out layout of hardcoded values

peter-budo committed:
[WIP] Award details:
 • Added setAlpha method to BaseDialogFragment
 • AwardGridFragment code formatting and 7 inch tablet detection
 • MedalAward added description and string IDs
 • Provided class for RibbonAwards to parse description and string ID
 • Dialog fragment already show some of data

peter-budo committed:
[WIP] Award details
 • BaseDialogFragment added set text method with drawable
 • Added missing expansion award resources
 • Added award expansion requirement
 • Updated changelog, includig missing assignments item

peter-budo committed:
Removed unused expansion colours, previously proposed for assignments expansion pack requirements

peter-budo committed:
Merge branch 'dev' into award-details

peter-budo committed:
Merge branch 'dev' into award-details

karllindmark commented on BF4Intel/src/main/res/values/styles.xml:
```diff
@@ -234,14 +234,14 @@
     <item name="android:textAppearance">@android:style/TextAppearance.Holo.Large</item>
     <item name="android:fontFamily">sans-serif-light</item>
-    <item name="android:textColor">#707070</item>
+    <item name="android:textColor">@color/regular_card_title</item>
```
:+1:

karllindmark commented on BF4Intel/src/main/res/layout/fragment_award_details.xml:
```diff
@@ -57,25 +44,38 @@
+      style="@style/NonListViewCard"
+      android:id="@+id/expansion_award_container">
+
+      <TextView
```
This should probably be centered vertically. :-)

karllindmark commented on BF4Intel/src/main/res/layout/fragment_award_details.xml:
```diff
@@ -83,9 +83,10 @@
-        android:layout_centerInParent="true"
         android:layout_alignParentTop="true"
-        android:text="@string/dummy_ribbon_1" />
+        android:textColor="@android:color/holo_blue_light"
```
We should possibly align this with the other blue colors currently used for titles, see the link below for more information (one example I remembered off the top of my head):

https://github.com/ninetwozero/com.ninetwozero.bf4intel/blob/dev/BF4Intel/src/main/res/layout/list_header_news_article.xml#L27

peter-budo replied:
That may be and issue as I have feeling that dialog title on tablet is holo light blue instead of dark blue

karllindmark replied:
We'll look into it in the future. :+1:

peter-budo committed:
• Fixed layout issue with missing layout width
 • Centered award extension requirement
 • Applied opacity on medal image only instead of whole container

karllindmark commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/awards/AwardDetailFragment.java:
```diff
@@ -18,19 +26,99 @@ public static AwardDetailFragment newInstance(final Bundle data) {
+
+    private void populateRibbonViews(final View view) {
+        setText(view, R.id.ribbon_title, AwardStringMap.get(award.getRibbon().getRibbonAward().getUniqueName()));
+        setAlpha(view, R.id.award_ribbon_container, award.getRibbon().isTaken() ? 1f : 0.5f);
```
This should probably be adjusting the alpha on the image, as for the medals above - currently the whole card is a bit opaque. :-)

karllindmark commented:
You can merge it once the alpha has been adjusted for the ribbon card. :+1:

peter-budo committed:
• Applied opacity to ribbon image only
 • Removed empty line

peter-budo committed:
• Reformatted code from assignments based on CI reports
 • Optimized imports
